### PR TITLE
Implement Wallet.listTransactions()

### PR DESCRIPTION
**Pull request by rorp**
It also fixes the compiler warning in tests.

### CODE REVIEW

**Review by benthecarman — APPROVED** [review]

File: testkit/src/test/scala/org/bitcoins/testkit/ImplicitsTest.scala [hunk 1]
```diff
@@ -19,12 +19,13 @@ class ImplicitsTest extends BitcoinSUnitTest {
...
     } catch {
-      case e: TestFailedException =>
+      case _: TestFailedException =>
         succeed
-      case e: Throwable => fail
+      case _: Throwable => fail
     }
   }
 
+  /* TODO fix the deprecation warning here
```
Comment (Christewart) [inline testkit/src/test/scala/org/bitcoins/testkit/ImplicitsTest.scala:28]: Why is this commetned out?

File: zmq/src/test/scala/org/bitcoins/zmq/ZMQSubscriberTest.scala [hunk 2]
```diff
@@ -38,7 +38,7 @@ class ZMQSubscriberTest extends AsyncFlatSpec {
     val socket = new InetSocketAddress("tcp://127.0.0.1", port)
 
     val context = ZMQ.context(1)
-    val publisher = context.socket(ZMQ.PUB)
+    val publisher = context.socket(SocketType.PUB)
```
Comment (Christewart) [inline zmq/src/test/scala/org/bitcoins/zmq/ZMQSubscriberTest.scala:41]: is ZMQ broken without this change?

File: wallet-test/src/test/scala/org/bitcoins/wallet/UTXOLifeCycleTest.scala [hunk 3]
```diff
@@ -32,20 +32,25 @@ class UTXOLifeCycleTest extends BitcoinSWalletTest {
...
+      oldTransactions <- wallet.listTransactions()
       tx <- wallet.sendToAddress(testAddr,
                                  Satoshis(3000),
                                  Some(SatoshisPerByte(Satoshis(3))))
 
       updatedCoins <- wallet.spendingInfoDAO.findOutputsBeingSpent(tx)
+      newTransactions <- wallet.listTransactions()
     } yield {
       assert(updatedCoins.forall(_.state == TxoState.PendingConfirmationsSpent))
+      assert(!oldTransactions.map(_.transaction).contains(tx))
```
Comment (Christewart) [inline wallet-test/src/test/scala/org/bitcoins/wallet/UTXOLifeCycleTest.scala:44]: nit: maybe use `.forall()` here? and everywhere else this pattern is repeated in the file

**Review by Christewart — COMMENTED** [review]
Oops wrong place

**Pull request merged**